docs/resources/connection_mysql.md

```diff
@@ -0,0 +1,229 @@
+#     HOST 'instance.foo000.us-west-1.rds.amazonaws.com',
+#     PORT 3306,
+#     USER SECRET "database"."schema"."example"
+#     PASSWORD SECRET "database"."schema"."example",
+# );
+
+
+# Create a MySQL Connection with SSH tunnel & plain text user
+resource "materialize_connection_mysql" "example_mysql_connection" {
+  name = "example_mysql_connection"
+  host = "instance.foo000.us-west-1.rds.amazonaws.com"
+  port = 3306
+
+  user {
+    text = "my_user"
+  }
+  password {
+    name          = "example"
+    database_name = "database"
+    schema_name   = "schema"
+  }
+  ssh_tunnel {
+    name = "example"
+  }
+}
+
+# CREATE CONNECTION example_mysql_connection TO POSTGRES (
+#     HOST 'instance.foo000.us-west-1.rds.amazonaws.com',
+#     PORT 3306,
+#     USER "my_user",
+#     PASSWORD SECRET "database"."schema"."example",
+#     SSH TUNNEL "example"
+# );
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `host` (String) The MySQL database hostname.
+- `name` (String) The identifier for the connection.
+- `user` (Block List, Min: 1, Max: 1) The MySQL database username.. Can be supplied as either free text using `text` or reference to a secret object using `secret`. (see [below for nested schema](#nestedblock--user))
+
+### Optional
+
+- `aws_privatelink` (Block List, Max: 1) The AWS PrivateLink configuration for the MySQL database. (see [below for nested schema](#nestedblock--aws_privatelink))
+- `comment` (String) **Public Preview** Comment on an object in the database.
+- `database_name` (String) The identifier for the connection database. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `ownership_role` (String) The owernship role of the object.
+- `password` (Block List, Max: 1) The MySQL database password. (see [below for nested schema](#nestedblock--password))
+- `port` (Number) The MySQL database port.
+- `region` (String) The region to use for the resource connection. If not set, the default region is used.
+- `schema_name` (String) The identifier for the connection schema. Defaults to `public`.
+- `ssh_tunnel` (Block List, Max: 1) The SSH tunnel configuration for the MySQL database. (see [below for nested schema](#nestedblock--ssh_tunnel))
+- `ssl_certificate` (Block List, Max: 1) The client certificate for the MySQL database.. Can be supplied as either free text using `text` or reference to a secret object using `secret`. (see [below for nested schema](#nestedblock--ssl_certificate))
+- `ssl_certificate_authority` (Block List, Max: 1) The CA certificate for the MySQL database.. Can be supplied as either free text using `text` or reference to a secret object using `secret`. (see [below for nested schema](#nestedblock--ssl_certificate_authority))
+- `ssl_key` (Block List, Max: 1) The client key for the MySQL database. (see [below for nested schema](#nestedblock--ssl_key))
+- `ssl_mode` (String) The SSL mode for the MySQL database.
```

Comment:
valid options are `disabled`, `required`, `verify-ca`, and `verify-identity`. These correspond to the same options on the MySQL client: https://dev.mysql.com/doc/refman/8.0/en/connection-options.html#option_general_ssl-mode

Reply:
Good point, I added this as a validate func as well!